pkg/runtime.v2/framework/plugins/torch/torch.go

```diff
@@ -43,11 +48,87 @@ func (t *Torch) Name() string {
 	return Name
 }
 
-func (t *Torch) EnforceMLPolicy(info *runtime.Info) error {
-	if info == nil || info.MLPolicy == nil || info.MLPolicy.Torch == nil {
+// TODO (andreyvelich): Add support for PyTorch elastic when JobSet supports Elastic Jobs.
+func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *kubeflowv2.TrainJob, runtimeMLPolicy *kubeflowv2.MLPolicy) error {
+	if info == nil || runtimeMLPolicy == nil || runtimeMLPolicy.Torch == nil {
 		return nil
 	}
-	// TODO: Need to implement main logic.
+
+	// TrainJob contains the actual information for the Trainer.
+	numNodes := runtimeMLPolicy.NumNodes
+	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumNodes != nil {
+		numNodes = trainJob.Spec.Trainer.NumNodes
+	}
+	info.Trainer.NumNodes = numNodes
```

Comment:
should we check if elastic policy is defined before setting numNodes?

Reply:
Discussion here: https://github.com/kubeflow/training-operator/pull/2308#discussion_r1823522840